pkg/controller/machineset/machineset_controller.go

```diff
@@ -494,12 +494,9 @@ func (r *ReconcileMachineSet) waitForMachineDeletion(machineList []*clusterv1.Ma
 func (r *ReconcileMachineSet) MachineToMachineSets(o handler.MapObject) []reconcile.Request {
 	result := []reconcile.Request{}
 
-	m := &clusterv1.Machine{}
-	key := client.ObjectKey{Namespace: o.Meta.GetNamespace(), Name: o.Meta.GetName()}
-	if err := r.Client.Get(context.Background(), key, m); err != nil {
-		if !apierrors.IsNotFound(err) {
-			klog.Errorf("Unable to retrieve Machine %q for possible MachineSet adoption: %v", key, err)
-		}
+	m, ok := o.Object.(*clusterv1.Machine)
```

Comment:
We should probably do similar in `pkg/controller/machinedeployment/machinedeployment_controller.go` for `MachineSetToDeployments`, but that can be a followup task to track. @ncdc thoughts?

Reply:
Yes, let's file issues for the others